### PR TITLE
[hybrid agent] add inspect otel subcommand

### PR DESCRIPTION
**Pull request by VihasMakwana**
## What does this PR do?

Add  a new `inspect otel` subcommand. This command outputs the hybrid mode config used by the elastic-agent. 

## Why is it important?

This subcommand is useful to get a quick snapshot of otel configuration without the need to "run diagnostics -> unzip -> go through config".

## Checklist

- [x] I have read and understood the [pull request guidelines](https://github.com/elastic/elastic-agent/blob/main/CONTRIBUTING.md#pull-request-guidelines) of this project.
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## How to test this PR locally

1. Build agent from this PR
2. Install it, or just run it using `./elastic-agent`
3. Run `elastic-agent inspect otel` to view the otel config

### CODE REVIEW

**Comment by cmacknz** [comment]
Thanks for this, I tried this using a policy that uses secrets paths to check the redaction. I found at least one bug, and I also think we shuold omit the monitoring components by default with an option to include them like standard inspect has. The collector config is quite a bit more verbose.

The agent policy I started with was:

```yaml
outputs:
  default:
    type: elasticsearch
    hosts: [127.0.0.1:9200]
    api_key: "example-key"
    #username: "elastic"
    #password: "changeme"
    preset: balanced

secret_paths: 
 - inputs.0.secret

inputs:
  - type: system/metrics
    id: unique-system-metrics-input
    data_stream.namespace: default
    use_output: default
    secret: redactme
    streams:
      - metricsets:
        - cpu
        data_stream.dataset: system.cpu
```

Which regular inspect gives:

```sh
❯ sudo elastic-development-agent inspect
agent:
  logging:
    to_stderr: true
inputs:
- data_stream:
    namespace: default
  id: unique-system-metrics-input
  secret: <REDACTED>
  streams:
  - data_stream:
      dataset: system.cpu
    metricsets:
    - cpu
  type: system/metrics
  use_output: default
outputs:
  default:
    api_key: <REDACTED>
    hosts:
    - 127.0.0.1:9200
    preset: balanced
    type: elasticsearch
secret_paths:
- inputs.0.secret
```

While otel inspect gives me the following where the `logs/_agent-component/system/metrics-default` isn't included at all so I can't verify the secret path redaction worked. 

```
❯ sudo elastic-development-agent inspect otel
[WARNING]: file may be partly redacted, could not cast value logs/_agent-component/system/metrics-default of type map[string][]string[WARNING]: file may be partly redacted, could not cast value logs/_agent-component/filestream-monitoring of type map[string][]string[WARNING]: file may be partly redacted, could not cast value logs/_agent-component/beat/metrics-monitoring of type map[string][]string[WARNING]: file may be partly redacted, could not cast value logs/_agent-component/http/metrics-monitoring of type map[string][]stringexporters:
  elasticsearch/_agent-component/default:
    api_key: <REDACTED>
    auth:
      authenticator: <REDACTED>
    compression: gzip
    compression_params:
...700 more lines
```

**Comment by cmacknz** [comment]
We also have an integration test for regular inspect and we should have one for otel inspect to cover redaction at a minimum:

https://github.com/elastic/elastic-agent/blob/8c787ddc9b9e2408af042befddc2715ad3d27608/testing/integration/ess/inspect_test.go#L26-L31

**Comment by VihasMakwana** [comment]
@cmacknz Thanks! I didn't cover the redaction part in the first draft of this PR. Looking at the existing inspect, it looks like a quick fix. I'll also add a similar test case to cover the scenarios. 

Moving it to draft for now.

File: internal/pkg/diagnostics/diagnostics.go [hunk 1]
```diff
@@ -387,6 +387,18 @@ func redactMap[K comparable](errOut io.Writer, inputMap map[K]interface{}) map[K
 	for rootKey, rootValue := range inputMap {
 		if rootValue != nil {
 			switch cast := rootValue.(type) {
+			case map[string][]string:
+				var newMap map[string]any = make(map[string]any, len(cast))
+				for k, v := range cast {
+					newMap[k] = v
+				}
+				rootValue = redactMap(errOut, newMap)
+			case []map[string]interface{}:
```
[the remark this answers is not in the document]
Comment (VihasMakwana) [reply internal/pkg/diagnostics/diagnostics.go]: I don't think this is accurate. In the subsequent `redactMap` call, the argument will be of type `map[string]any`. It will be get processed as expected.

Anyhow, this is a corner case to handle the `service::pipelines::*` section in otel config.

**Comment by VihasMakwana** [comment]
@cmacknz It seems secrets don’t work the same in otel mode. During conversion from normal components to beat receivers, some fields may be dropped based on their position.

For example, the secret field gets removed in this config when it’s converted to a beat receiver.
 

```yaml
inputs:
- data_stream:
    namespace: default
  id: unique-system-metrics-input
  secret: <REDACTED>
  streams:
  - data_stream:
      dataset: system.cpu
    metricsets:
    - cpu
  type: system/metrics
  use_output: default
outputs:
...
secret_paths:
- inputs.0.secret
```

Whereas, following would work fine and the secret is redacted:

```yaml
inputs:
- data_stream:
    namespace: default
  id: unique-system-metrics-input
  custom_attr: custom_value
  streams:
  - data_stream:
      dataset: system.cpu
    metricsets:
    - cpu123
    secret: xyz
  type: system/metrics
  use_output: default
outputs:
...
secret_paths:
- inputs.0.streams.0.secret
```

**Comment by VihasMakwana** [comment]
I have pushed a fix for optional monitoring and correctly redacting secrets. I'll mark this ready for review once I add an integration test case.

Feel free to test it out 😄

File: internal/pkg/diagnostics/diagnostics.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (VihasMakwana) [reply internal/pkg/diagnostics/diagnostics.go]: Fixed this.
